### ISSUE TITLE
Guard doesn't start _guard-core with the same Ruby version it was originally started

### ISSUE THREAD

**Issue by moll** [guard/guard#878]
Hey,

I suspect there's a bug in `bin/guard` due to the way it spawns `_guard-core`. Namely, with a `Gemfile` (example below) stating the Ruby version, Bundler may run `bin/guard` with a Ruby executable not named `ruby` — such as `/usr/bin/ruby24`. However, because `_guard-core` is executed as a regular file and it delegates to `/usr/bin/env`, the core process is started with the system default Ruby. Fixing it just requires prepending `spawn(*args)` with `Gem.ruby` for `spawn(Gem.ruby, *args)`.

Gemfile:
```
source "https://rubygems.org"
ruby "~> 2.3"
gem "guard"
```

```sh
bundle exec guard
```

Cheers

**Comment by moll** [comment]
I threw in a simple PR should my proposed fix be accepted.

**Comment by rymai** [comment]
That makes sense, but I guess your `$PATH` is not properly configured because `which ruby` should give you the current Ruby, not the system one?

For example:

```
› which ruby
/Users/remy/.rubies/ruby-2.3.4/bin/ruby
```

**Comment by moll** [comment]
Nope. This talks about a case when Ruby is _not_ named `ruby`. Which is on any platform that uses suffixes. ;)

**Comment by rymai** [comment]
> This talks about a case when Ruby is not named ruby. Which is on any platform that uses suffixes. ;)

Oh, good call! Ok, I guess `Gem.ruby` doesn't hurt then!
